django/db/backends/mysql/base.py

```diff
@@ -245,7 +238,7 @@ def init_connection_state(self):
             assignments.append('SQL_AUTO_IS_NULL = 0')
 
         if self.isolation_level:
-            assignments.append("%s = '%s'" % (self.transaction_isolation_variable, self.isolation_level))
+            assignments.append("SESSION TRANSACTION ISOLATION LEVEL %s" % self.isolation_level.upper())
 
         if assignments:
             with self.cursor() as cursor:
```

Comment:
this now generates invalidate SQL if both assignments happen:

```
chainz@localhost [7]> set session transaction isolation level read committed, sql_auto_is_null = 0;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'sql_auto_is_null = 0' at line 1
chainz@localhost [8]>
```

Instead we should generate two statements, they can be sent as a single sql string to avoid roundtrips though by using a `;` delim:

```
chainz@localhost [8]> set session transaction isolation level read committed; set sql_auto_is_null = 0;
Query OK, 0 rows affected (0.00 sec)

Query OK, 0 rows affected (0.00 sec)

chainz@localhost [9]>
```